core/tools.js

```diff
@@ -38,6 +38,81 @@
 			} else {
 				return namedEntities[ code ];
 			}
+		},
+		// this is the polyill for the CSS.escape for IE.
+		// https://github.com/mathiasbynens/CSS.escape/blob/master/css.escape.js
+		cssEscapeForIE = function( value ) {
+			var string = String(value);
+			var length = string.length;
+			var index = -1;
+			var codeUnit;
+			var result = '';
+			var firstCodeUnit = string.charCodeAt(0);
+			while ( ++index < length ) {
+				codeUnit = string.charCodeAt(index);
+				// Note: there’s no need to special-case astral symbols, surrogate
+				// pairs, or lone surrogates.
+
+				// If the character is NULL (U+0000), then the REPLACEMENT CHARACTER
+				// (U+FFFD).
+				if ( codeUnit == 0x0000 ) {
+					result += '\uFFFD';
+					continue;
+				}
+
+				if (
+					// If the character is in the range [\1-\1F] (U+0001 to U+001F) or is
+					// U+007F, […]
+					(codeUnit >= 0x0001 && codeUnit <= 0x001F) || codeUnit == 0x007F ||
```

Comment:
All of those range checks could be replaced with extracted helper function inside this function scope.
```js
function isInRange(value, min, max) {
				return value <= min && value >= max;
			}
```

Then all if statements could be simplified to `isInRange( codeUnit, 0x0030, 0x0039 )`. Of course with proper values, so in this case it will be:
```js
( isInRange( codeUnit, 0x0001, 0x001F  ) ) || codeUnit == 0x007F ||
```

After this change, it will be also easier to simplify comments and make the code a little bit more compact.

Finally it could be something like this:

```js
if (
	// If the character is in the range [\1-\1F] or is U+007F
	isInRange( codeUnit, 0x0001, 0x001F ) || codeUnit == 0x007F ||
	// If the character is the first character and is in the range [0-9]
	( index == 0 && isInRange( codeUnit, 0x0030, 0x0039 ) ) ||
	// If the character is the second character and is in the range [0-9]
	// and the first character is a `-` (U+002D)
	( index == 1 && isInRange( codeUnit, 0x0030, 0x0039 ) && firstCodeUnit == 0x002D )
) {
	// https://drafts.csswg.org/cssom/#escape-a-character-as-code-point
	result += '\\' + codeUnit.toString( 16 ) + ' ';
	continue;
}
```

or even without comments if we use more meaningful const values

```js
if ( isInRange( codeUnit, UTF16.SOH, UTF16.SEPARATOR ) || codeUnit == UTF16.DEL ||
	( index == 0 && isInRange( codeUnit, UTF16.ZERO, UTF16.NINE ) ) ||
	( index == 1 && isInRange( codeUnit, UTF16.ZERO, UTF16.NINE ) && firstCodeUnit == UTF16.HYPHEN )
) {
	// https://drafts.csswg.org/cssom/#escape-a-character-as-code-point
	result += '\\' + codeUnit.toString( 16 ) + ' ';
	continue;
}
```
Although I'm not so happy with const names for `0x0001` and `0x001F`. If you have any better idea or think they should stay as they are - please express that in the next commits.

Maybe still not great, but more readable.